### PR TITLE
Oauth pt2

### PR DESCRIPTION
**Pull request by jhgould**
Added Oauth to user dashboard. Will need to refactor for Omniauth but I have a working app now so lets roll with it until we are back on pace.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -25,7 +25,9 @@ gem 'active_model_serializers'
 gem 'omniauth-google-oauth2'
 gem 'will_paginate'
 gem 'acts-as-taggable-on', '~> 6.0'
-gem 'omniauth-census', git: "https://github.com/turingschool-projects/omniauth-census"
+# gem 'omniauth-census', git: "https://github.com/turingschool-projects/omniauth-census"
+gem 'omniauth-github', github: 'omniauth/omniauth-github', branch: 'master'
+# gem 'devise'
```
Comment (jhgould) [inline Gemfile:30]: Took out those two gems as they were not needed. I think I was getting an error due to the first omniauth gem.

File: app/controllers/admin/tutorials_controller.rb [hunk 2]
```diff
@@ -3,19 +3,19 @@ def edit
...
       create_videos(tutorial)
       # YoutubeService.new.playlist_video_params(tutorial)
       # YoutubeService.new.create_videos(tutorial)
       redirect_to admin_dashboard_path
-      flash[:notice] = "Successfully created tutorial. #{view_context.link_to("View it here.", tutorial_path(tutorial.id))}"
+      flash[:notice] = "Successfully created tutorial. #{view_context.link_to('View it here.', tutorial_path(tutorial.id))}"
     end
   end
-  
+
```
Comment (jhgould) [inline app/controllers/admin/tutorials_controller.rb:16]: Ran rubo cop. These are changes it made.

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -3,6 +3,31 @@ def new
...
+    auth = JSON.parse(oauth_response.body)
+    # user          = User.find_or_create_by(id: auth["id"])
+    user          = User.find(session[:user_id])
+    user.token    = "token #{token}"
+    # user.password = "password"
+    user.save!
+    session[:user_id] = user.id
+    redirect_to dashboard_path
+  end
+
```
Comment (jhgould) [inline app/controllers/sessions_controller.rb:30]: Followed the lesson plan word for word on Oauth. On line 24 I had to interpolate the token to have the correct format that GH needs.

File: spec/features/admin/new_tutorial_spec 2.rb [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    fill_in "tutorial_playlist_id",   with: "PL83DDC2327BEB616D" 
+    click_on 'Save' 
+
+    click_link('View it here.')
+    tutorial = Tutorial.last
+
+    expect(tutorial.videos.count).to be > 200
+  end
+end
+
```
Comment (jhgould) [inline spec/features/admin/new_tutorial_spec 2.rb:70]: I didnt add this in. Not sure where it came from or why its saying I added it.

File: app/controllers/sessions_controller.rb (hunk 3 above)
Comment (jhgould) [reply app/controllers/sessions_controller.rb:30]: Also something of note is the commented-out line on 25. I was getting an error at one point that `password` cannot be blank. I commented-out the line later and everything was working but if you get a validation error that is probably the cause.

File: spec/features/admin/new_tutorial_spec 2.rb (hunk 4 above)
Comment (muydanny) [reply spec/features/admin/new_tutorial_spec 2.rb:70]: This is a spec I wrote

**Pull request merged**